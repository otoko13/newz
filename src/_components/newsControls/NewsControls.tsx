import React from 'react';
import { Stack } from 'office-ui-fabric-react/lib/Stack';
import './newsControls.scss';
import { ActionButton } from 'office-ui-fabric-react/lib/Button';
import NewstypeFilter from './NewsTypeFilter';

export interface INewsControlsProps {
    onRefreshClick: () => void;
    filters: string[];
    onFiltersChanged: (types: string[]) => void;
}

const NewsControls = (props: INewsControlsProps) => {
    return (
        <Stack horizontal verticalAlign='center' horizontalAlign='space-between' className='NewsControls'>
            <Stack.Item>
                <NewstypeFilter onFiltersChanged={props.onFiltersChanged} filters={props.filters} />
            </Stack.Item>
            <Stack.Item>
                <div>
                    <ActionButton className='refresh-button' aria-label='refresh' onClick={props.onRefreshClick} iconProps={{iconName: 'Refresh'}} />
                </div>
            </Stack.Item>
        </Stack>
    );
}

export default NewsControls;