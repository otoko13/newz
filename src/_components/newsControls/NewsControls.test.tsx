import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import NewsControls, { INewsControlsProps } from './NewsControls';

let props: INewsControlsProps;

beforeEach(() => {
    props = {
        onRefreshClick: jest.fn(),
        onStoryTypeChange: jest.fn(),
        storiesType: 'new',
        filters: ['story'],
        onFiltersChanged: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<NewsControls {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('changing filters', () => {
    it('should call props.onFiltersChanged with the correct new news type filters', () => {
        const { getAllByRole, getAllByTitle } = render(<NewsControls {...props} />);
        const storyFilter = getAllByRole('button')[0];
        fireEvent.click(storyFilter);
        fireEvent.click(getAllByTitle('Stories')[1]);
        fireEvent.click(getAllByTitle('Polls')[1]);
        fireEvent.click(getAllByTitle('Jobs')[1]);
        expect(props.onFiltersChanged).toHaveBeenCalledWith(['story', 'job']);
    });

    it('should call props.onStoryTypeChange with the correct new stories type', () => {
        const { getByRole, getByTitle } = render(<NewsControls {...props} />);
        fireEvent.click(getByRole('listbox'));
        fireEvent.click(getByTitle('Best stories'));
        expect(props.onStoryTypeChange).toHaveBeenCalledWith('best');
    });
});

describe('refreshing', () => {
    it('should call props.onRefreshClick when clicked', () => {
        const { getByLabelText } = render(<NewsControls {...props} />);
        expect(props.onRefreshClick).not.toHaveBeenCalled();
        fireEvent.click(getByLabelText('refresh'));
        expect(props.onRefreshClick).toHaveBeenCalled();
    });
});