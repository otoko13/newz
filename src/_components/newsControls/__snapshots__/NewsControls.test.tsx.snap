// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ms-Stack NewsControls css-40"
>
  <div
    class="ms-StackItem css-41"
  >
    <div
      class="ms-Dropdown-container NewsTypeFilter root-42"
    >
      <label
        class="ms-Label ms-Dropdown-label label-60"
        id="Dropdown0-label"
      >
        What do you want to see?
      </label>
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="Dropdown0-label Dropdown0-option"
        class="ms-Dropdown dropdown-44"
        data-is-focusable="true"
        id="Dropdown0"
        role="button"
        tabindex="0"
      >
        <span
          aria-atomic="true"
          aria-invalid="false"
          aria-live="polite"
          class="ms-Dropdown-title title-45"
          id="Dropdown0-option"
        >
          Stories
        </span>
        <span
          class="ms-Dropdown-caretDownWrapper caretDownWrapper-46"
        >
          <i
            aria-hidden="true"
            class="ms-Dropdown-caretDown caretDown-61"
            data-icon-name="ChevronDown"
          />
        </span>
      </div>
    </div>
    <div
      class="ms-Dropdown-container StoryTypeFilter root-42"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="ms-Dropdown dropdown-44"
        data-is-focusable="true"
        id="Dropdown1"
        role="listbox"
        tabindex="0"
      >
        <span
          aria-atomic="true"
          aria-invalid="false"
          aria-live="polite"
          aria-posinset="1"
          aria-selected="true"
          aria-setsize="3"
          class="ms-Dropdown-title title-45"
          id="Dropdown1-option"
          role="option"
        >
          Newest
        </span>
        <span
          class="ms-Dropdown-caretDownWrapper caretDownWrapper-46"
        >
          <i
            aria-hidden="true"
            class="ms-Dropdown-caretDown caretDown-61"
            data-icon-name="ChevronDown"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    class="ms-StackItem css-41"
  >
    <div>
      <button
        aria-label="refresh"
        class="ms-Button ms-Button--icon refresh-button root-62"
        data-is-focusable="true"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-63"
          data-automationid="splitbuttonprimary"
        >
          <i
            aria-hidden="true"
            class="ms-Icon root-33 ms-Button-icon icon-65"
            data-icon-name="Refresh"
            role="presentation"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
